integration/src/main/java/scripts/utils/ClientTestUtils.java

```diff
@@ -239,7 +244,7 @@ public static boolean jobIsRunning(JobReport jobReport) {
    * @throws InterruptedException
    */
   public static @Nullable <T> T getWithRetryOnException(
-      Supplier<T> supplier,
+      SupplierWithException<T> supplier,
```

Comment:
does this serve to restrict the allowed suppliers? Or is it just for documentation?

Reply:
The default Supplier does not have a throws clause, so cannot be used with any code that throws a checked exception.